### PR TITLE
O3-1459: Create E2E automated workflow to test patient conditions recoring feature in o3

### PR DESCRIPTION
**Pull request by pasindur99**
## Purpose
The pull request is to create patients' condition record E2E test for O3. And this pull request is related to the issue ticket number [O3-1459](https://issues.openmrs.org/browse/O3-1459)

## Goals.
- Check the status of the patient's condition record End to End process.

## Approach
 - Create a feature file and a step definition file for the test.
 - Create GitHub workflow.
 - Migrate the test result to the dashboard.

### CODE REVIEW

**Comment by pasindur99** [comment]
Patient registration is failing due to an issue with the front end. (as what we can expect in the E2E test). So the pr can be reviewed and merged.
cc: @jayasanka-sack  @ibacher

File: packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/12-patient-conditions/patient-conditions.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Given("the user is logged in", () => {
+  cy.login();
+});
+
+Given("the user arrives on a patient’s summary page", () => {
+  cy.visit(`patient/${patient.uuid}/chart`);
+});
+
+When("the user clicks on Conditions tab", () => {
+  cy.contains("Conditions").click({ force: true });
```
Comment (jayasanka-sack) [inline packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/12-patient-conditions/patient-conditions.js:24]: Any specific reasons to force click?

Comment (jayasanka-sack) [reply packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/12-patient-conditions/patient-conditions.js:24]: The comment applies to other force clicks too.

File: packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/02-patient-registration/patient-registration.js [hunk 2]
```diff
@@ -57,27 +57,28 @@ When('the user enters {string} details for Andria Faiza', validity => {
...
     if (user.phoneNumber != null) {
         cy.getByLabel('Phone number(optional)').type(user.phone, {force: true});
     }
-    // Click on the first day on the calendar because the calendar doesn't support manual inputs
-    cy.getByLabel('Date of Birth').click({force: true});
-    cy.get('.dayContainer .flatpickr-day').first().click({force: true});
+  // Manually input the date
+  cy.getByLabel('Date of Birth').click({force: true}).type("01/01/2019");
+  // Click on the first day on the calendar because the calendar doesn't support manual inputs
+  // cy.get('.dayContainer .flatpickr-day').first().click({force: true});
```
Comment (jayasanka-sack) [inline packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/02-patient-registration/patient-registration.js:77]:
```suggestion
```

**Comment by jayasanka-sack** [comment]
> Patient registration is failing due to an issue with the front end. (as what we can expect in the E2E test). So the pr can be reviewed and merged. cc: @jayasanka-sack @ibacher

@pasindur99 Could you please create a ticket explaining the frontend issue?

File: packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/12-patient-conditions/patient-conditions.js (hunk 1 above)
Comment (pasindur99) [reply packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/12-patient-conditions/patient-conditions.js:24]: We can use the `click()`method without force click. But randomly cypress won't click the tabs in the patient chart for some reason. @jayasanka-sack

File: packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/02-patient-registration/patient-registration.js (hunk 2 above)
Comment (pasindur99) [reply packages/cypress/cypress/integration/cucumber/step_definitions/refapp-3.x/02-patient-registration/patient-registration.js:77]: The issue in patient registration is with adding the birthday of the patient. Until it gets fixed I think it's good to keep the old method as well in the test workflow file. @jayasanka-sack

**Comment by pasindur99** [comment]
I will create a ticket.

**Review by jayasanka-sack — APPROVED** [review]
Thanks @pasindur99 !

**Pull request merged**